### ISSUE TITLE
MacOS Instructions Unclear

### ISSUE THREAD

**Issue by Puddinglord** [MaterialFoundry/MaterialDeck#43]
In Windows, it is very possible to double click something and have a server run. On macOS, if you double click the "document" in the folder it tries to open itself with TextEdit. I propose that you just update the wiki to include the simple instructions to run the server manually. They would be as follows.

1. clone the repo
2. cd into the cloned project
3. run `npm install` to make sure all the dependencies are up to date
4. run the server by using the command `node MaterialServer.js`

You could also keep releasing the server and providing slightly different instructions if you just wanted to release the server as a .zip of the source code to still keep things simple.

**Comment by CDeenen** [comment]
If you download the macos archive from here: https://github.com/CDeenen/MaterialServer/releases, you shouldn't need to clone the repository or use node. On my old Macbook I could start the server by doubleclicking the MaterialServer-macos file in that archive. Alternatively you should be able to run it by dragging that file into the terminal.

Could you please check if that works, I'll then edit the documentation to describe the correct method.

**Comment by Puddinglord** [comment]
What version of macOS are you running on your Macbook? I am running Big Sur. Double-clicking defaults to TextEdit and dragging it to the terminal says command not found.

**Comment by CDeenen** [comment]
El Capitan (unfortunately the latest OSX I can run on the machine), but someone else also had success running it on a newer version. I'm waiting on a response on what OSX version they're running.

**Comment by wonk** [comment]
@Puddinglord I have not made it to Big Sur which may have changed things as it's the first full revision of OSX/MacOS.  However, on Catalina, I do the following 3-step to run the server.
1)  Download MacOS version from URL, above.
2)  Double-click the zip to extract it to its own folder, "MaterialServer-macos".
3)  Just inside the “MaterialServer-macos” folder, double-click the "Unix executable" named "MaterialServer-macos".

Step #3 runs the executable in Terminal.app.  Perhaps right-click the executable to “Open in...” the executable in Terminal.app explicitly (to overrride the default app, e.g., TextEdit.app)?  (If that works, perhaps change the default app for that type of file (in Get Info) to open in Terminal.app by default?)

**Comment by DM-miX** [comment]
Using Catilina 10.15.5 it's not running for me.
@wonk Have you got something else installed or some settings that might let it run?

Double-click
- Set Terminal as app to open it.
- Opens terminal but no window appears. If a terminal window is already open nothing happens


Drag MaterialServer-macos to termianl
- Terminal says Permission Denied
- Add sudo to the front: Command not Available


Directly in terminal
- Navigate to the MaterServer folder
- Type MaterialServer-macos: 'Command not available'

**Comment by wonk** [comment]
@DM-miX :  Nothing I can think of though I'll poke around.  I am running 10.15.7 Catalina, in case that makes any difference.  More after I poke around.

**Comment by wonk** [comment]
@DM:  When you 'Get Info' on the file "MaterialServer-macos", does it list as "Kind: Unix executable"?

**Comment by wonk** [comment]
@DM:  Terminal.app has a lot of options.  I don't recall tweaking them, so I believe mine are default.  Are yours default or tweaked?  In any case, my options are vanilla, so I let Terminal.app open without any special shell or such.

If you have any ideas what else I could check, I am happy to help.  For now, I'm not seeing anything out of the ordinary with my setup.

**Comment by DM-miX** [comment]
I've had a win! The term 'Unix executable' lead me on the right path

1. Open Terminal and navigate to the MaterialServer-macos folder
2. type: chmod 755 MaterialServer-macos (you _might_ need to add sudo at the start)
3. From Finder - Double-click MaterialServer-macos. It will open in terminal, but you'll get "midi.node can't opened because it's from unidentified developer" warning. Close terminal.
4. Open System Preferences > Security & Privacy. Look in the lower half for "midi.node" and click the "Allow" button
5. Again doble-click MaterialServer-macos. This time you'll get ANOTHER warning, click open
6. Now you're up and running - hopefully with no more warnings

**Comment by CDeenen** [comment]
Thanks for all the responses. It looks like there's a solution. I'll expand the documentation with some of this info.
I'll keep this issue open for a few more days, and if there are no more problems I'll close it.

**Comment by wonk** [comment]
@DM-miX Glad to you got it working.  Kudos!

**Comment by Tetraplasm** [comment]
> I've had a win! The term 'Unix executable' lead me on the right path
> 
> 1. Open Terminal and navigate to the MaterialServer-macos folder
> 2. type: chmod 755 MaterialServer-macos (you _might_ need to add sudo at the start)
> 3. From Finder - Double-click MaterialServer-macos. It will open in terminal, but you'll get "midi.node can't opened because it's from unidentified developer" warning. Close terminal.
> 4. Open System Preferences > Security & Privacy. Look in the lower half for "midi.node" and click the "Allow" button
> 5. Again doble-click MaterialServer-macos. This time you'll get ANOTHER warning, click open
> 6. Now you're up and running - hopefully with no more warnings

I know this issue is closed, but I'm having trouble importing the StreamDeck profile on Catalina 10.15.7. I've tried downloading the Profile v1.2.2 as well as v1.2.1 and the Kuvek and XL versions (mine is an XL) and importing them by double clicking the files (has worked for other profiles I've imported in the past) and also opening the Stream Deck software and forcing it to import the file using the menus there. I set up the server just the way you described it here and it worked, so I'm wondering if you know what may be causing this problem. I can open a separate issue if necessary.

**Comment by CDeenen** [comment]
You're not the first person who has reported problems with importing profiles, besides the profiles, there was also someone who had trouble installing the plugin. I don't know why this happens, it all seems to work fine for most people, but for some reason not for everyone. If I remember correctly, you're the 2nd person reporting this issue with OSX, but there was also someone on Windows with the same problem.
Considering this is functionality provided by the SD application, I don't think this is an issue related to MD, but a bug in the SD application. So you might have better luck seeking support somewhere else.
There a subreddit for profiles: https://www.reddit.com/r/streamdeckprofiles/
And a general Elgato subreddit: https://www.reddit.com/r/elgato/
They might be able to help you out better than I can. If it ends up being something I could fix, please open a new issue for it.
